dataset/dataset.py

```diff
@@ -65,44 +59,65 @@ def __init__(self, name, env):
         self.pair_attr_stats = {}
 
-    # Fixed to load data from a CSV file at the moment.
-    def load_data(self, name, f_path, f_name, na_values=None):
+    # TODO(richardwu): load more than just CSV files
+    def load_data(self, name, fpath, na_values=None, entity_col=None, src_col=None):
         """
         load_data takes a CSV file of the initial data, adds tuple IDs (_tid_)
         to each row to uniquely identify an 'entity', and generates unique
         index numbers for each attribute/column.
 
         Creates a table with the user supplied 'name' parameter (e.g. 'hospital').
-        """
 
+        :param name: (str) name to initialize dataset with.
+        :param fpath: (str) filepath to CSV file.
+        :param na_values: (str) value that identifies a NULL value
+        :param entity_col: (str) column containing the unique
+            identifier/ID of an entity.  For fusion tasks, rows with
+            the same ID will be fused together in the output.
+            If None, assumes every row is a unique entity.
+        :param src_col: (str) if not None, for fusion tasks
+            specifies the column containing the source for each "mention" of an
+            entity.
+        """
         tic = time.clock()
         try:
-            # Load raw CSV file/data into the Postgres 'init_X' table.
+            # Do not include TID and source column as trainable attributes
+            exclude_attr_cols = ['_tid_']
+            if src_col is not None:
+                exclude_attr_cols.append(src_col)
+
+            # Load raw CSV file/data into a Postgres table 'name' (param).
+            self.raw_data = Table(name, Source.FILE, na_values=na_values, exclude_attr_cols=exclude_attr_cols, fpath=fpath)
 
-            self.raw_data = Table(name, Source.FILE, f_path, f_name, na_values)
-            # Add _tid_ column to dataset
             df = self.raw_data.df
-            df.insert(0,'_tid_', range(0,len(df)))
-            df.fillna('_nan_',inplace=True)
+            # Add _tid_ column to dataset that uniquely identifies an entity.
+            # If entity_col is not supplied, use auto-incrementing values.
+            # Otherwise we use the entity values directly as _tid_'s.
+            if entity_col is None:
+                # auto-increment
+                df.insert(0, '_tid_', range(0,len(df)))
+            else:
+                # use entity IDs as _tid_'s directly
+                df.rename({entity_col: '_tid_'}, axis='columns', inplace=True)
+
+            # Use '_nan_' to represent NULL values
+            df.fillna('_nan_', inplace=True)
+
+            logging.info("Loaded %d rows with %d cells", self.raw_data.df.shape[0], self.raw_data.df.shape[0] * self.raw_data.df.shape[1])
+
             # Call to store to database
             self.raw_data.store_to_db(self.engine.engine)
-            status = 'DONE Loading '+f_name
+            status = 'DONE Loading {fname}'.format(fname=os.path.basename(fpath))
```

Comment:
Nit: Seems like both str.format() and % are used to format strings in the codebase